### DISCUSSION TITLE
Lesson 6: Error while deploying the contract (SimpleStorage.sol)

### DISCUSSION THREAD

**Discussion by 14himanshi** [Cyfrin/foundry-full-course-cu#1909] [Q&A] [answered]
I am using windows-wsl and ganache for deploying my contract (SimpleStorage ). I am getting this error for a long time now , can anyone please help me :)
The code for SimpleStorage contract is the same one as the one shown in the YouTube video . This was the first time i tried to deploy the contract 

[Doubt.pdf](https://github.com/user-attachments/files/16045337/Doubt.pdf)

**Comment by EngrPips** [comment]
Hello @14himanshi, Why don't you deploy to `Anvil` instead of `Ganache`?

**Reply by 14himanshi** [reply]
Hello EngrPips,

First of all, I want to express my gratitude for your reply. I was worried that no one would respond, so I'm very happy to hear from you!

I have tried deploying the SimpleStorage contract using Anvil as well, but I am still encountering the following errors:

[Doubt1.pdf](https://github.com/user-attachments/files/16045414/Doubt1.pdf)

Your assistance would be greatly appreciated in helping me resolve this issue. :)

**Reply by EngrPips** [reply]
You are welcome. Cyfrin will never leave you hanging. Know that for sure

**Reply by EngrPips** [reply]
@14himanshi, are you closing the terminal where you got the `Anvil` information? You are supposed to leave it running while you add the network to your metamask. Also, you don't need to add it to your metamask to be able to deploy to `Anvil` on your terminal

**Reply by EngrPips** [reply]
You can also share images instead of PDFs. It makes the process of helping you easier.

**Reply by 14himanshi** [reply]
![image](https://github.com/Cyfrin/foundry-full-course-cu/assets/102136102/f7a78695-9c96-46c6-b553-27c19a78378e)

I was actually doing ctrl+z as soon as I got the output. So I should not do ctrl+z and just type the next set of instructions on new window?
Also, i didn't get what you were trying to say here: " Also, you don't need to add it to your metamask to be able to deploy to Anvil on your terminal"

**Reply by EngrPips** [reply]
Yeah don't exit the Anvil process, instead open a new terminal and run the command there. I saw you were trying to add Anvil network to your Metmask so I was saying that's not needed to deploy to the Anvil Local chain. Just keep the Anvil process running then open another terminal where you will run the deployment command and your contract will be deployed

**Comment by yasharyan0904** [comment]
Better use Anvil:
1)Run Anvil in one terminal
2)Open new terminal, then $forge create SimpleStorage --rpc-url http://127.0.0.1:8545 --private-key yourANVILkey

**Reply by 14himanshi** [reply]
Hello yasharyan0904,
Thank you for replying :)
I tried using anvil but this is the error I am getting
[Doubt2.pdf](https://github.com/user-attachments/files/16045458/Doubt2.pdf)
Thank you in advance for your time and assistance. :)

**Reply by yasharyan0904** [reply, marked as answer]
Look, just open the terminal and type Anvil, let it run don't close it.
Open new terminal and type->forge create SimpleStorage --rpc-url http://127.0.0.1:8545/ --private-key yourANVILkey
You will get the the AnvilPRIVATEkey from the running anvil !!

**Reply by EngrPips** [reply]
@14himanshi, Please do as @yasharyan0904 suggested, and let's see how that plays out.

**Reply by 14himanshi** [reply]
It worked :)
Thank you very much yasharyan0904 and EngrPpis :)

**Reply by EngrPips** [reply]
Awesome, glad you unblocked yourself. Keep crushing it.
